src/org/exist/jetty/ServerShutdown.java

```diff
@@ -20,83 +14,81 @@
+import se.softhouse.jargo.ArgumentException;
+import se.softhouse.jargo.CommandLineParser;
+import se.softhouse.jargo.ParsedArguments;
 
 import java.io.IOException;
 import java.io.InputStream;
 
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.util.List;
 import java.util.Properties;
 
+import static org.exist.util.ArgumentUtil.getOpt;
+import static se.softhouse.jargo.Arguments.helpArgument;
+import static se.softhouse.jargo.Arguments.stringArgument;
+
 /**
  * Call the main method of this class to shut down a running database instance.
  * 
  * @author wolf
  */
 public class ServerShutdown {
 
-    // command-line options
-    private final static int HELP_OPT = 'h';
+    /* general arguments */
+    private static final Argument<?> helpArg = helpArgument("-h", "--help");
+
+    /* database connection arguments */
+    private static final Argument<String> userArg = stringArgument("-u", "--user")
+            .description("specify username (has to be a member of group dba).")
+            .required()
+            .defaultValue("admin")
+            .build();
+    private static final Argument<String> passwordArg = stringArgument("-p", "--password")
+            .description("specify password for the user.")
+            .required()
+            .defaultValue("")
+            .build();
+    private static final Argument<String> uriArg = stringArgument("-u", "--uri")
+            .description("the XML:DB URI of the database instance to be shut down.")
+            .build();
 
-    private final static int URI_OPT = 'l';
-
-    private final static int USER_OPT = 'u';
-
-    private final static int PASSWORD_OPT = 'p';
+    @SuppressWarnings("unchecked")
+	public static void main(final String[] args) {
+        try {
+            final ParsedArguments arguments = CommandLineParser
+                    .withArguments(userArg, passwordArg, uriArg)
+                    .andArguments(helpArg)
+                    .parse(args);
 
-    private final static CLOptionDescriptor OPTIONS[] = new CLOptionDescriptor[] {
-            new CLOptionDescriptor("help",
-                    CLOptionDescriptor.ARGUMENT_DISALLOWED, HELP_OPT,
-                    "print help on command line options and exit."),
-            new CLOptionDescriptor("user",
-                    CLOptionDescriptor.ARGUMENT_REQUIRED, USER_OPT,
-                    "specify username (has to be a member of group dba)."),
-            new CLOptionDescriptor("password",
-                    CLOptionDescriptor.ARGUMENT_REQUIRED, PASSWORD_OPT,
-                    "specify password for the user."),
-            new CLOptionDescriptor("uri", CLOptionDescriptor.ARGUMENT_REQUIRED,
-                    URI_OPT,
-                    "the XML:DB URI of the database instance to be shut down.") };
+            process(arguments);
+        } catch (final ArgumentException e) {
+            System.out.println(e.getMessageAndUsage());
+            System.exit(2);
```

Comment:
`128` is reserved for when you give an invalid number (i.e. `0 > n > 255`) to `exit`